### PR TITLE
chore: update Go toolchain to v1.16.3

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:

See https://github.com/deis/docker-go-dev/releases/tag/v1.31.0

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

This fixes two new warnings emitted by `shellcheck`:

  https://www.shellcheck.net/wiki/SC2269 -- This variable is assigned to itself
  https://www.shellcheck.net/wiki/SC2268 -- Avoid x-prefix in comparisons

### CODE REVIEW

File: scripts/get-akse.sh [hunk 1]
```diff
@@ -79,7 +79,7 @@ checkDesiredVersion() {
   elif type "wget" > /dev/null; then
     TAG=$(wget -q -O - $release_url | awk '/\/tag\//' | grep -v no-underline | grep "<a href=\"/Azure/aks-engine/releases" | head -n 1 | cut -d '"' -f 2 | awk '{n=split($NF,a,"/");print a[n]}' | awk 'a !~ $0{print}; {a=$0}')
   fi
-  if [ "x$TAG" == "x" ]; then
```
Comment (jackfrancis) [inline scripts/get-akse.sh:82]: sneaking this one in? :)

File: test/e2e/cluster.sh [hunk 2]
```diff
@@ -159,8 +159,6 @@ else
     SKIP_AFTER_UPGRADE="${SKIP_AFTER_SCALE_DOWN}|${SKIP_AFTER_UPGRADE}"
   elif [ "${SCALE_CLUSTER}" = "true" ]; then
     SKIP_AFTER_UPGRADE="${SKIP_AFTER_SCALE_DOWN}"
-  else
```
Comment (jackfrancis) [inline test/e2e/cluster.sh:162]: I see what you did here

File: scripts/get-akse.sh (hunk 1 above)
Comment (mboersma) [reply scripts/get-akse.sh:82]: This release of docker-go-dev also bumped `shellcheck` and it had new two new complaints which I fixed. This one is [SC2268](https://github.com/koalaman/shellcheck/wiki/SC2268), which was a workaround for ancient, non-POSIX-compliant shells that it suggests is just confusing today.

File: test/e2e/cluster.sh (hunk 2 above)
Comment (mboersma) [reply test/e2e/cluster.sh:162]: This is [SC2269](https://github.com/koalaman/shellcheck/wiki/SC2269): "This variable is assigned to itself, so the assignment does nothing."

File: scripts/get-akse.sh (hunk 1 above)
Comment (jackfrancis) [reply scripts/get-akse.sh:82]: Well I lived long enough to see shellcheck take one for the team in the interest of practicality.

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Pull request merged**